src/components/backup/BackupConfirmPasswordStep.js

```diff
@@ -0,0 +1,285 @@
+    `
+  : styled(Column)`
+      ${borders.buildRadius('top', 16)};
+      background-color: ${colors.white};
+      height: 100%;
+    `;
+
+const Container = styled(Column)`
+  background-color: ${colors.transparent};
+  height: 100%;
+`;
+
+const Shadow = styled(ShadowStack).attrs({
+  borderRadius: 16,
+  height: 49,
+  shadows: [
+    [0, 10, 30, colors.dark, 0.1],
+    [0, 5, 15, colors.dark, 0.04],
+  ],
+  width: Dimensions.get('window').width - 130,
+})`
+  elevation: 15;
+  margin-bottom: 19;
+`;
+
+const InputsWrapper = styled(View)`
+  align-items: center;
+  height: 120;
+`;
+
+const PasswordInput = styled(Input).attrs({
+  blurOnSubmit: false,
+  letterSpacing: 0.2,
+  secureTextEntry: true,
+  size: 'large',
+  weight: 'normal',
+})`
+  padding-bottom: 15;
+  padding-left: 19;
+  padding-right: 40;
+  padding-top: 15;
+`;
+
+const IconWrapper = styled(View)`
+  margin-bottom: 12;
+  height: 22;
+  position: absolute;
+  right: 12;
+  top: 12;
+  width: 22;
+`;
+
+const Title = styled(Text).attrs({
+  size: 'big',
+  weight: 'bold',
+})`
+  margin-bottom: 12;
+`;
+
+const DescriptionText = styled(Text).attrs({
+  align: 'center',
+  color: colors.alpha(colors.blueGreyDark, 0.5),
+  lineHeight: 'looser',
+  size: 'large',
+})`
+  padding-bottom: 30;
+  padding-left: 50;
+  padding-right: 50;
+`;
+
+const WarningIcon = () => (
+  <IconWrapper>
+    <Icon color={colors.orangeLight} name="warningCircled" size={22} />
+  </IconWrapper>
+);
+
+const TopIcon = () => (
+  <GradientText
+    align="center"
+    angle={false}
+    colors={['#FFB114', '#FF54BB', '#00F0FF']}
+    end={{ x: 0, y: 0 }}
+    letterSpacing="roundedTight"
+    size={52}
+    start={{ x: 1, y: 1 }}
+    steps={[0, 0.5, 1]}
+    weight="bold"
+  >
+    <Text size={52}>􀙶</Text>
+  </GradientText>
+);
+
+const BackupConfirmPasswordStep = () => {
+  const { goBack } = useNavigation();
+  const { params } = useRoute();
+  const dispatch = useDispatch();
+  const [validPassword, setValidPassword] = useState(false);
+  const [incorrectPassword, setIncorrectPassword] = useState(false);
+  const [passwordFocused, setPasswordFocused] = useState(true);
+  const [password, setPassword] = useState('');
+  const [label, setLabel] = useState('􀎽 Confirm Backup');
+  const passwordRef = useRef();
+  const { latestBackup, wallets } = useWallets();
+
+  useEffect(() => {
+    const fetchPasswordIfPossible = async () => {
+      const pwd = await fetchBackupPassword();
+      if (pwd) {
+        setPassword(pwd);
+      }
+    };
+
+    fetchPasswordIfPossible();
+  }, []);
+
+  const onPasswordFocus = useCallback(() => {
+    setPasswordFocused(true);
+  }, []);
+
+  const onPasswordBlur = useCallback(() => {
+    setPasswordFocused(false);
+  }, []);
+
+  useEffect(() => {
+    let newLabel = '';
+    let passwordIsValid = false;
+
+    if (incorrectPassword) {
+      newLabel = 'Incorrect Password';
+    } else {
+      if (password !== '' && password.length >= 8) {
```

Comment:
- Perhaps we should include some trimming before checking if empty
- Also noticed this logic existing in multiple places, maybe nice to have a simple util function for password validity